### PR TITLE
Fixed #16501 -- Added a unicode parameter to models.SlugField and forms.SlugField.

### PR DESCRIPTION
**Pull request by berkerpeksag**
https://code.djangoproject.com/ticket/16501

### CODE REVIEW

**Comment by berkerpeksag** [comment]
> I believe we also need to include its license file similar to how we do for jQuery. A selenium test for the JavaScript changes would also be good to add.

Done.

File: django/contrib/admin/options.py [hunk 1]
```diff
@@ -642,7 +642,7 @@ def media(self):
         if self.actions is not None:
             js.append('actions%s.js' % extra)
         if self.prepopulated_fields:
-            js.extend(['urlify.js', 'prepopulate%s.js' % extra])
+            js.extend(['xregexp.min.js', 'urlify.js', 'prepopulate%s.js' % extra])
```
Comment (timgraham) [inline django/contrib/admin/options.py]: Needs a rebase (these files are now alway included as of 28e8c54d7d2800a8ac98cf646a9e074055bc1b1b).

File: django/contrib/admin/templates/admin/prepopulated_fields_js.html [hunk 2]
```diff
@@ -8,7 +8,8 @@
         id: '#{{ field.field.auto_id }}',
         dependency_ids: [],
         dependency_list: [],
-        maxLength: {{ field.field.field.max_length|default_if_none:"50"|unlocalize }}
+        maxLength: {{ field.field.field.max_length|default_if_none:"50"|unlocalize }},
+        unicode: {{ field.field.field.unicode|lower }},
```
Comment (timgraham) [inline django/contrib/admin/templates/admin/prepopulated_fields_js.html]: I think a trailing comma creates a problem on some browsers (IE).

**Comment by timgraham** [comment]
I wonder if we might call the parameter `ascii_only` or `allow_unicode` to be a bit more explicit.

**Comment by berkerpeksag** [comment]
+1 to `allow_unicode`.

**Comment by timgraham** [comment]
Updated in https://github.com/django/django/pull/4518

**Pull request closed without merging**